cockatrice/src/releasechannel.cpp

```diff
@@ -0,0 +1,307 @@
+    bool ok;
+    QString tmp = QString(reply->readAll());
+    reply->deleteLater();
+
+    QVariantMap resultMap = QtJson::Json::parse(tmp, ok).toMap();
+    if (!ok) {
+        qDebug() << "No reply received from the tag update server";
+        emit error(tr("No reply received from the tag update server."));
+        return;
+    }
+
+    if(!(resultMap.contains("object") && 
+        resultMap["object"].toMap().contains("sha"))) {
+        qDebug() << "Invalid received from the tag update server";
+        emit error(tr("Invalid reply received from the tag update server."));
+        return;
+    }
+
+    lastRelease->setCommitHash(resultMap["object"].toMap()["sha"].toString());
+    qDebug() << "Got reply from tag server, size=" << tmp.size()
+        << "commit=" << lastRelease->getCommitHash();
+
+    qDebug() << "Searching for a corresponding file on the stable channel: " << QString(STABLEFILES_URL);
+    response = netMan->get(QNetworkRequest(QString(STABLEFILES_URL)));
+    connect(response, SIGNAL(finished()), this, SLOT(fileListFinished()));
+}
+
+void StableReleaseChannel::fileListFinished()
+{
+    QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
+    bool ok;
+    QString tmp = QString(reply->readAll());
+    reply->deleteLater();
+
+    QVariantList resultList = QtJson::Json::parse(tmp, ok).toList();
+    if (!ok) {
+        qDebug() << "No reply received from the file update server";
+        emit error(tr("No reply received from the file update server."));
+        return;
+    }
+
+    QString shortHash = lastRelease->getCommitHash().left(GIT_SHORT_HASH_LEN);
+    QString myHash = QString(VERSION_COMMIT);
+    qDebug() << "Current hash=" << myHash << "update hash=" << shortHash;
+
+    bool needToUpdate = (QString::compare(shortHash, myHash, Qt::CaseInsensitive) != 0);
+    bool compatibleVersion = false;
+
+    foreach(QVariant file, resultList)
+    {
+        QVariantMap map = file.toMap();
+        // TODO: map github version to bintray version
+        /*
+        if(!map.contains("version"))
+            continue;
+        if(!map["version"].toString().endsWith(shortHash))
+            continue;
+        */
+
+        if(!downloadMatchesCurrentOS(map))
+            continue;
+
+        compatibleVersion = true;
+
+        QString url = QString(STABLEDOWNLOAD_URL) + map["path"].toString();
+        lastRelease->setDownloadUrl(url);
+        qDebug() << "Found compatible version url=" << url;
+        break;
+    }
+    
+    emit finishedCheck(needToUpdate, compatibleVersion, lastRelease);
+}
+
+DevReleaseChannel::DevReleaseChannel(int _index, QString _name)
+:ReleaseChannel(_index, _name)
+{
+}
+
+QString DevReleaseChannel::getManualDownloadUrl()
+{
+    return QString(DEVMANUALDOWNLOAD_URL);
+}
+
+void DevReleaseChannel::checkForUpdates()
+{
+    qDebug() << "Searching for updates on the dev channel: " << DEVRELEASE_URL;
+    response = netMan->get(QNetworkRequest(QString(DEVRELEASE_URL)));
```

Comment:
This code is almost identical to the other class - it seems like url and log message are the only difference. Might be good to make it a method of the base interface they both extend and take the urls as parameters